parcel.go

```diff
@@ -2,59 +2,121 @@ package main
 
 import (
 	"database/sql"
+	"errors"
 )
 
+// Parcel представляет информацию о посылке
+type Parcel struct {
+	Number    int
+	Client    int
+	Status    string
+	Address   string
+	CreatedAt string
+}
+
+// ParcelStatusRegistered представляет статус зарегистрированной посылки
+const ParcelStatusRegistered = "registered"
+
+// ParcelStore представляет хранилище посылок
 type ParcelStore struct {
 	db *sql.DB
 }
 
+// NewParcelStore создает новый экземпляр ParcelStore с переданной базой данных
 func NewParcelStore(db *sql.DB) ParcelStore {
 	return ParcelStore{db: db}
 }
 
+// Add добавляет новую посылку в базу данных
 func (s ParcelStore) Add(p Parcel) (int, error) {
-	// реализуйте добавление строки в таблицу parcel, используйте данные из переменной p
-
-	// верните идентификатор последней добавленной записи
-	return 0, nil
+	query := `INSERT INTO parcel (client, status, address, created_at) VALUES (?, ?, ?, ?)`
+	result, err := s.db.Exec(query, p.Client, p.Status, p.Address, p.CreatedAt)
+	if err != nil {
+		return 0, err
+	}
+	id, err := result.LastInsertId()
+	if err != nil {
+		return 0, err
+	}
+	return int(id), nil
 }
 
+// Get возвращает посылку по заданному номеру
 func (s ParcelStore) Get(number int) (Parcel, error) {
-	// реализуйте чтение строки по заданному number
-	// здесь из таблицы должна вернуться только одна строка
-
-	// заполните объект Parcel данными из таблицы
-	p := Parcel{}
-
+	query := `SELECT number, client, status, address, created_at FROM parcel WHERE number = ?`
+	row := s.db.QueryRow(query, number)
+	var p Parcel
+	err := row.Scan(&p.Number, &p.Client, &p.Status, &p.Address, &p.CreatedAt)
+	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
Для сравнивания ошибок нужно использовать специальную функцию
```suggestion
		if errors.Is(err,  sql.ErrNoRows) {
```